src/hobbits-plugins/importerexporters/UsbDevice/usbdevice.cpp

```diff
@@ -0,0 +1,458 @@
+            if (progress->isCancelled())
+            { // check if the user has cancelled the import operation
+                break;
+            }
+            progress->setProgress(i, transferNum);                                 //update the progressbar
+            std::this_thread::sleep_for(std::chrono::milliseconds(transferDelay)); //wait for the durration of the transfer delay
+        }
+        params = exitLibusb(true, params); //exit libusb with closing the device
+        break;
+    case 3:
+        /**
+         * This is the same thing as the bulk transfer implementation, but instead with an interrup transfer, only 2 lines of
+         * code are different, these lines replace bulk transfer with interrupt transfer.
+         * Refer to the bulk transfer implementation for a thorough explanation.
+         * 
+         */
+
+        transferTypeStr += ", Interrupt Transfer";
+        params = setupLibusb(params);
+
+        if (params.errorCode < 0)
+        {
+            return returnError(params.errorCode);
+        }
+
+        for (int i = 0; i < transferNum; i++)
+        {
+            if (libusb_kernel_driver_active(params.handle, params.interfaceNum) == 1)
+            {
+                attach = true;
+                libusb_detach_kernel_driver(params.handle, params.interfaceNum);
+            }
+            int transferReturn = libusb_interrupt_transfer(params.handle, params.endpoint, smallBuffer, (int)sizeof(smallBuffer), &actualLength, transferTimeout);
+            if (transferReturn == 0 && actualLength != 0)
+            {
+                frames->appendRange(transferSize * 8);
+                for (int j = 0; j < transferSize; j++)
+                {
+                    largeBuffer.append(smallBuffer[j]);
+                }
+            }
+            else
+            {
+                if (transferReturn != LIBUSB_ERROR_TIMEOUT)
+                {
+                    if (attach)
+                    {
+                        libusb_reset_device(params.handle);
+                        attach = false;
+                    }
+                    return ImportResult::error("Transfer Error, Code: " + QString::number(transferReturn) + " " + libusb_error_name(transferReturn));
+                }
+                else
+                {
+                    if (includeTimeout)
+                    {
+                        largeBuffer.append("TIMEOUT");
+                        frames->appendRange(56);
+                    }
+                }
+            }
+            if (attach)
+            {
+                libusb_reset_device(params.handle);
+                attach = false;
+            }
+            if (progress->isCancelled())
+            {
+                break;
+            }
+            progress->setProgress(i, transferNum);
+            std::this_thread::sleep_for(std::chrono::milliseconds(transferDelay));
+        }
+        params = exitLibusb(true, params);
+        break;
+
+    default:
```

Comment:
shouldn't there be an error if it's an unexpected type? Also, any compelling reason to keep this as a switch statement?